src/server/config.rs

```diff
@@ -295,6 +331,30 @@ impl Config {
     }
 }
 
+pub struct ServerConfigManager(pub Arc<VersionTrack<Config>>);
+
+impl ConfigManager for ServerConfigManager {
+    fn dispatch(&mut self, c: ConfigChange) -> std::result::Result<(), Box<dyn std::error::Error>> {
```

Comment:
We can also send a `Task::RefreshConfigEvent` to the snap worker, so we don't need `register_refresh_cfg_event`.